### PR TITLE
[HybridKV][Bugfix] Fix Hybrid kvcache sharing bug in same attention type

### PR DESCRIPTION
**Pull request by MengqingCao**
### What this PR does / why we need it?
Part of https://github.com/vllm-project/vllm-ascend/pull/3106
Fix Hybrid kvcache sharing bug in same attention type
Change the `shared_by` logic so that the same attention spec could share the same buffer instead of allocating more hbm.
After this pr, kvcache memory saved 50% in qwen3-next compared with before (`self_attn:linear_attn=1:3` in an `attn_group`), and `gpu_memory_utilization` could increase to `0.8` on Qwen3-Next when running on A2 64G/card with tp4

image

### How was this patch tested?
Test pass with the latest e2e test case on qwen3-next

- vLLM version: v0.11.0rc3
- vLLM main: https://github.com/vllm-project/vllm/commit/c9461e05a4ed3557cfbf4b15ded1e26761cc39ca

### CODE REVIEW

**Comment by wangxiyuan** [comment]
any relationship with this PR https://github.com/vllm-project/vllm-ascend/pull/3404/ ?

File: vllm_ascend/worker/model_runner_v1.py [hunk 1]
```diff
@@ -3154,25 +3175,26 @@ def initialize_kv_cache_tensors(
...
-                        else:
-                            cache_size_aligned = kv_cache_tensor.size + alignment
-                            tensor = torch.zeros(cache_size_aligned,
-                                                 dtype=torch.int8,
-                                                 device=self.device)
-                            tensor = self._align_memory(
-                                tensor, alignment)[:kv_cache_tensor.size]
-                        kv_cache_raw_tensors[layer_name_inner] = tensor
-                elif "attn" in layer_name:
+                        # shared the kvcache between the self_attn specs in the same group
```
Comment (MengqingCao) [inline vllm_ascend/worker/model_runner_v1.py]:
```suggestion
                        # shared the kvcache between the linear_attn specs in the same group
```

**Comment by MengqingCao** [comment]
> any relationship with this PR #3404 ?

We both attempt to fix kvcache allocation duplication issue, but I keep allocating buffer for different attn type seperately to avoid accuracy issue

**Review by yiz-liu — APPROVED** [review]
